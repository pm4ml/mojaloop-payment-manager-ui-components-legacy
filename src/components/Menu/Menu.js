import './Menu.scss';
import '../../icons/mule/back-small.svg';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import { NamedIcon } from '../Common';
import Icon from '../Icon';

const getPathMatches = (pathname, path, partial) => {
  if (!path) {
    return false;
  }
  if (partial && pathname.startsWith(path)){
    return true;
  }
  if (path === pathname) {
    return true;
  }

  const pathChunks = path.split('/');
  const pathNameChunks = pathname.split('/');

  if (pathChunks.length !== pathNameChunks.length) {
    return false;
  }

  return pathNameChunks.every((chunk, index) => {
    const pathChunk = pathChunks[index];
    const isExact = pathChunk === chunk;
    const isWild = pathChunk !== undefined && pathChunk.startsWith(':');
    return isExact || isWild;
  });
  
};

const bindOnClickProp = onClick => element =>
  React.cloneElement(element, {
    ...element.props,
    onClick,
  });

const bindPathnameProp = pathname => element =>
  React.cloneElement(element, {
    ...element.props,
    pathname,
  });

const bindActiveProp = pathname => element => {
  const { path, partial, back, active } = element.props;
  const pathMatches = getPathMatches(pathname, path, partial) && !back;
  if (pathMatches || active) {
    return React.cloneElement(element, {
      ...element.props,
      active: true,
    });
  }
  return element;
};
const bindKey = (element, index) =>
  React.cloneElement(element, {
    ...element.props,
    key: index.toString(),
  });

const bindDisabledProp = disabled => element =>
  React.cloneElement(element, {
    ...element.props,
    disabled: element.props.disabled || disabled,
  });

class MenuItem extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
  onClick() {
    if (!this.props.disabled) {
      this.props.onClick(this.props.to || this.props.path);
    }
  }
  render() {
    const { label, icon, fill, size, disabled, hidden, active, back, namedIcon } = this.props;
    if (hidden) {
      return null;
    }
    let backIcon = null;
    if (back) {
      backIcon = <Icon className="el-menu__item__back-icon" name="arrow" size={10} fill="#999" />;
    }
    let itemIcon = null;

    if (icon) {
      itemIcon = (
        <div className="el-menu__item__item-icon">
          <Icon className="el-menu__item__icon" name={icon} size={size} fill={fill} />
        </div>
      );
    } else if (namedIcon) {
      itemIcon = (
        <div className="el-menu__item__item-named-icon">
          <NamedIcon namedIconTitle={label} />
        </div>
      );
    }
    const classNames = utils.composeClassNames([
      'el-menu__item',
      active && 'el-menu__item--active',
      disabled && 'el-menu__item--disabled',
      back && 'el-menu__item--back',
      icon && 'el-menu__item--with-icon',
    ]);

    return (
      <div className={classNames} onClick={this.onClick} role="presentation">
        {backIcon}
        {itemIcon}
        {label}
      </div>
    );
  }
}

MenuItem.defaultProps = {
  path: undefined,
  to: undefined,
  disabled: false,
  hidden: false,
  back: false,
  icon: undefined,
  namedIcon: false,
  fill: undefined,
  size: 14,
  asRoot: false,
  partial: false,
};
MenuItem.propTypes = {
  path: PropTypes.string,
  to: PropTypes.string,
  disabled: PropTypes.bool,
  hidden: PropTypes.bool,
  back: PropTypes.bool,
  icon: PropTypes.string,
  namedIcon: PropTypes.bool,
  fill: PropTypes.string,
  size: PropTypes.number,
  // prop asRoot is not used directly by the MenuItem but it is used by the Menu component
  // eslint-disable-next-line react/no-unused-prop-types
  asRoot: PropTypes.bool,
  // prop partial is not used directly by the MenuItem but it is used by the Menu component
  // eslint-disable-next-line react/no-unused-prop-types
  partial: PropTypes.bool,
};

const MENU_ITEM_TYPE = (<MenuItem />).type;

const isMenuItem = node => node.type === MENU_ITEM_TYPE;

const MenuSection = ({
  pathname,
  label,
  children,
  onClick,
  hidden,
  disabled,
  icon,
  size,
  fill,
}) => {
  if (hidden) {
    return null;
  }
  const menuItems = React.Children.toArray(children)
    .filter(element => isMenuItem(element))
    .map(bindOnClickProp(onClick))
    .map(bindActiveProp(pathname))
    .map(bindDisabledProp(disabled));

  let itemIcon = null;
  if (icon) {
    itemIcon = (
      <div className="el-menu__section-icon">
        <Icon name={icon} size={size} fill={fill} />
      </div>
    );
  }

  let menuSectionLabel = null;
  if (label) {
    menuSectionLabel = (
      <div className="el-menu__section-label">
        {itemIcon}
        {label}
      </div>
    );
  }

  const classNames = utils.composeClassNames([
    'el-menu__section',
    disabled && 'el-menu__section--disabled',
  ]);

  return (
    <div className={classNames}>
      {menuSectionLabel}
      <div className="el-menu__section-items">{menuItems}</div>
    </div>
  );
};

MenuSection.defaultProps = {
  label: undefined,
  icon: undefined,
  fill: '#7C7C7C',
  disabled: false,
  size: 14,
};
MenuSection.propTypes = {
  label: PropTypes.string,
  icon: PropTypes.string,
  fill: PropTypes.string,
  size: PropTypes.number,
  disabled: PropTypes.bool,
};

const MENU_SECTION_TYPE = (<MenuSection />).type;

const isMenuSection = node => node.type === MENU_SECTION_TYPE;

const MenuItemsGroup = ({ children }) => <div className="el-menu__section-items">{children}</div>;

const wrapItemsInSections = items => {
  const groupedMenuItems = [];
  let currentGroup = [];
  const addCurrenntGroupToGroupedItems = () => {
    if (currentGroup.length) {
      groupedMenuItems.push(<MenuItemsGroup>{currentGroup}</MenuItemsGroup>);
      currentGroup = [];
    }
  };

  items.forEach(node => {
    if (isMenuSection(node) || node.props.back) {
      addCurrenntGroupToGroupedItems();
      groupedMenuItems.push(node);
    } else {
      currentGroup.push(node);
    }
  });

  addCurrenntGroupToGroupedItems();
  return groupedMenuItems.map(bindKey);
};

class Menu extends PureComponent {
  static flattenMenuSections(items) {
    // It flattens nested MenuSection, MenuItem components into a one-level array
    return items.reduce(
      (prevItems, currentItem) => [
        ...prevItems,
        ...(isMenuSection(currentItem)
          ? Menu.flattenMenuSections(React.Children.toArray(currentItem.props.children))
          : [currentItem]),
      ],
      [],
    );
  }
  constructor() {
    super();
    this.getActiveNode = this.getActiveNode.bind(this);
  }
  getActiveNode(parentNode) {
    const { pathname } = this.props;
    let activeNode = null;
    // render Menu when pathname matches path
    if (getPathMatches(pathname, parentNode.props.path, parentNode.props.partial)) {
      return parentNode;
    }
    // Default to Menu when going manual - no route matching
    if (parentNode === this && parentNode.props.pathname === undefined) {
      activeNode = parentNode;
    }

    // Flatten MenuSections in order not to have nested children when detecting active menu
    const items = Menu.flattenMenuSections(React.Children.toArray(parentNode.props.children));

    items.some(node => {
      // find the first matching menu item and return the parent or the item itself
      // depending if needs to be treated like a root
      if (isMenuItem(node)) {
        const { path, asRoot, partial, children, active } = node.props;
        const pathMatches = getPathMatches(pathname, path, partial);

        if (pathMatches || active) {
          // asRoot prop is meant to be used when menu has child elements
          // and we do not want to render the parent node but the child nodes
          activeNode = asRoot ? node : parentNode;
        } else if (children !== undefined) {
          activeNode = this.getActiveNode(node);
        }
      }
      return activeNode;
    });

    return activeNode;
  }

  render() {
    const { pathname, onChange } = this.props;
    let menuComponents = null;
    const activeNode = this.getActiveNode(this);

    if (activeNode !== null) {
      menuComponents = React.Children.toArray(activeNode.props.children);
      menuComponents = wrapItemsInSections(
        menuComponents
          .filter(element => isMenuItem(element) || isMenuSection(element))
          .map(bindOnClickProp(onChange))
          .map(bindPathnameProp(pathname))
          .map(bindActiveProp(pathname)),
      );
    }
    return <div className="mb-element el-menu">{menuComponents}</div>;
  }
}

Menu.defaultProps = {
  path: undefined,
  pathname: undefined,
};
Menu.propTypes = {
  // prop path is not used directly but it is used in getActiveNode
  // eslint-disable-next-line react/no-unused-prop-types
  path: PropTypes.string,
  pathname: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default Menu;
export { MenuItem, MenuSection };
