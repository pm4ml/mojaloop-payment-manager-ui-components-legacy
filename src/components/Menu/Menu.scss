@import '../../assets/styles/vars/colors.scss';

.el-menu {
  overflow: hidden;
  background: #f2f1f1;
}

.el-menu__section {
}

.el-menu__section-icon {
  padding-right: 10px;
}

.el-menu__section-label {
  text-align: center;
  width: 100%;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: left;
  padding-left: 10px;
  background: $color__white;
  margin-top: 1px;
}

.el-menu__section--disabled {
  opacity: 0.5;
}

.el-menu__section-items {
  margin-top: 1px;
  margin-bottom: 1px;
  background: $color__gray-1;
}

.el-menu__item {
  display: flex;
  align-items: center;
  width: 100%;
  padding-left: 16px;
  min-height: 30px;
  cursor: pointer;
  line-height: 19px;
  border-left: 4px solid transparent;
  transition: border 0.2s linear, color 0.2s linear, fill 0.2s linear;
  font-size: 12px;

  &:hover:not(.el-menu__item--active) {
    color: $color__main-primary;

    .el-menu__item__icon {
      fill: $color__main-primary;
    }
  }
}

.el-menu__item__icon {
  transition: fill 0.2s linear;
  fill: #a7a7a7;
}

.el-menu__item--active {
  background: #f2f1f1;
  font-weight: bold;
  color: #333;

  .el-menu__item__icon {
    fill: $color__dark-blue-5;
  }
}

.el-menu__item--disabled {
  color: #999;
  fill: #ccc;
  border-color: transparent;
}

.el-menu__item--back {
  padding-left: 0px;
  height: 40px;
  background: $color__white;
  margin-top: 1px;
}

.el-menu__item--with-icon {
}

.el-menu__item__back-icon {
  transform: rotateZ(180deg);
  margin-left: 3px;
  margin-right: 3px;
}

.el-menu__item__item-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16px;
  height: 100%;
  margin-right: 10px;
  padding-bottom: 2px;
}

.el-menu__item__item-named-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  margin-right: 10px;
  padding-bottom: 2px;
}
