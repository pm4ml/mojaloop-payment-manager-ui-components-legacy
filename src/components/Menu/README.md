# Menu 

Renders a Menu. Composed by the components
  - [Menu](#menu-component)
  - [MenuSection](#menusection-component)
  - [MenuItem](#menuitem-component)


## Menu component 

Renders the whole menu component 

### Proptypes

|   Name   |   Type   |           Description           | Optional | Values |   Default   |
|----------|----------|---------------------------------|----------|--------|-------------|
| path     | `string` | describes the starting path     |          |        | `undefined` |
| pathname | `string` | describes the current pathname  |          |        | `undefined` |
| onChange | `func`   | describes the callback function |          |        | `undefined` |


## MenuSection component

Renders a menu section with the label

### Proptypes

|  Name |   Type   |         Description         | Optional | Values |   Default   |
|-------|----------|-----------------------------|----------|--------|-------------|
| label | `string` | describes the section label |          |        | `undefined` |

## MenuItem Component

Renders a menu item

### Proptypes

|   Name   |   Type   |                 Description                 | Optional | Values |   Default   |
|----------|----------|---------------------------------------------|----------|--------|-------------|
| path     | `string` | describes the path where the item is active |          |        | `undefined` |
| to       | `string` | describes the path to go to                 |          |        | `undefined` |
| disabled | `bool`   | describes if the item is disabled           |          |        | `false`     |
| hidden   | `bool`   | describes if should be hidden               |          |        | `false`     |
| back     | `bool`   | describes if it renders as a _back_ button  |          |        | `false`     |
| icon     | `string` | describes the icon name to render           |          |        | `undefined` |
| fill     | `string` | describes the icon fill color               |          |        | `undefined` |
| size     | `number` | describes the icon size                     |          |        | `undefined` |
