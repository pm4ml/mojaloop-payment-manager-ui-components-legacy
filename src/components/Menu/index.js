import Menu, { MenuItem, MenuSection } from './Menu';

export default Menu;
export { MenuItem, MenuSection };
