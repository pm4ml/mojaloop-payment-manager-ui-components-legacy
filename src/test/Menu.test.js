import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import Menu, { MenuItem, MenuSection } from '../components/Menu';

const onChangeMockEvent = jest.fn();

it('renders only the menu', () => {
  const wrapper = mount(<Menu path="/" pathname="/" onChange={onChangeMockEvent} />);
  expect(wrapper.find('.el-menu')).toHaveLength(1);
});

it('renders the menu wrapper also if pathname does not match any route', () => {
  const wrapper = mount(<Menu path="/" pathname="/test" onChange={onChangeMockEvent} />);
  expect(wrapper.find('.el-menu')).toHaveLength(1);
});

it('renders the menu items if parent root matches', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo" />
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  expect(wrapper.find('.el-menu__item')).toHaveLength(2);
});

it('renders the menu items if child route matches', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo" />
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  expect(wrapper.find('.el-menu__item')).toHaveLength(2);
});

it('renders the menu section with icons', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuSection label="Menu Section" icon="access-manager-color">
        <MenuItem path="/foo" label="foo" />
        <MenuItem path="/bar" label="bar" />
      </MenuSection>
    </Menu>,
  );
  expect(wrapper.find('.el-menu__section-icon')).toHaveLength(1);
});

it('renders the menu items with icons', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuSection label="Menu Section">
        <MenuItem path="/foo" label="foo" icon="admin-color" />
        <MenuItem path="/bar" label="bar" icon="application-color" />
      </MenuSection>
    </Menu>,
  );
  expect(wrapper.find('.el-menu__item__item-icon')).toHaveLength(2);
});

it('renders the menu items with named icon', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuSection label="Menu Section">
        <MenuItem path="/foo" label="foo" namedIcon />
        <MenuItem path="/bar" label="bar" namedIcon />
      </MenuSection>
    </Menu>,
  );
  expect(wrapper.find('.el-menu__item__item-named-icon')).toHaveLength(2);
});

it('does not render the menu items if no route matches', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/non-existing" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo" />
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  expect(wrapper.find('.el-menu__item')).toHaveLength(0);
});

it('renders the nested child when asRoot prop is set and route matches parent', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo" asRoot>
        <MenuItem path="/foo/nested" label="nested" />
      </MenuItem>
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  const menuItem = wrapper.find('.el-menu__item');
  expect(menuItem).toHaveLength(1);
  expect(menuItem.text()).toBe('nested');
  expect(wrapper.find('.el-menu__item')).toHaveLength(1);
});

it('renders the parent child if asRoot prop is not set and route matches parent', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo">
        <MenuItem path="/foo/nested" label="nested" />
      </MenuItem>
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  const menuItems = wrapper.find('.el-menu__item');
  expect(menuItems).toHaveLength(2);
  expect(wrapper.find('.el-menu__item')).toHaveLength(2);
  expect(menuItems.at(0).text()).toBe('foo');
  expect(menuItems.at(1).text()).toBe('bar');
});

it('renders the nested child if matching if asRoot prop is not set ', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/1st/2nd/3rd/4th" onChange={onChangeMockEvent}>
      <MenuItem path="/1st" label="1st Level">
        <MenuItem path="/1st/2nd" label="2st level">
          <MenuItem path="/1st/2nd/3rd" label="3nd level">
            <MenuItem path="/1st/2nd/3rd/4th" label="4th level" />
          </MenuItem>
        </MenuItem>
      </MenuItem>
    </Menu>,
  );

  const menuItem = wrapper.find('.el-menu__item');
  expect(menuItem).toHaveLength(1);
  expect(menuItem.text()).toBe('4th level');
});

it('renders the menu items when partial match is allowed', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo/partial/match" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo" partial />
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  const menuItem = wrapper.find('.el-menu__item');
  const activeMenuItem = wrapper.find('.el-menu__item--active');
  expect(menuItem).toHaveLength(2);
  expect(activeMenuItem.text()).toBe('foo');
});

it('renders the menu items when router param is used', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/users/:user-id/settings" onChange={onChangeMockEvent}>
      <MenuItem path="/books" label="Books" />
      <MenuItem path="/users/:user-id" label="User" asRoot>
        <MenuItem path="/users/:user-id/account" label="account" />
        <MenuItem path="/users/:user-id/settings" label="settings" />
      </MenuItem>
    </Menu>,
  );
  const menuItem = wrapper.find('.el-menu__item');
  const activeMenuItem = wrapper.find('.el-menu__item--active');
  expect(menuItem).toHaveLength(2);
  expect(activeMenuItem.text()).toBe('settings');
});

it('renders the menu section if parent root matches', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/" onChange={onChangeMockEvent}>
      <MenuSection label="Menu Section">
        <MenuItem path="/foo" label="foo" />
        <MenuItem path="/bar" label="bar" />
      </MenuSection>
    </Menu>,
  );
  const menuSection = wrapper.find('.el-menu__section');
  const menuSectionLabel = menuSection.find('.el-menu__section-label');
  expect(menuSectionLabel).toHaveLength(1);
  expect(menuSectionLabel.text()).toBe('Menu Section');
});

it('renders the Menu when no pathname is set', () => {
  const wrapper = mount(
    <Menu onChange={onChangeMockEvent}>
      <MenuItem label="foo" />
      <MenuItem label="bar" />
    </Menu>,
  );
  const menuItems = wrapper.find('.el-menu__item');
  expect(menuItems).toHaveLength(2);
});

it('renders the manually set active prop', () => {
  const wrapper = mount(
    <Menu onChange={onChangeMockEvent}>
      <MenuItem label="foo" active />
      <MenuItem label="bar" />
    </Menu>,
  );
  const activeMenuItem = wrapper.find('.el-menu__item--active');
  expect(activeMenuItem).toHaveLength(1);
  expect(activeMenuItem.text()).toBe('foo');
});

it('renders the disabled prop on menu item', () => {
  const wrapper = mount(
    <Menu onChange={onChangeMockEvent}>
      <MenuItem label="foo" disabled />
      <MenuItem label="bar" />
    </Menu>,
  );
  const disabledMenuItem = wrapper.find('.el-menu__item--disabled');
  expect(disabledMenuItem).toHaveLength(1);
  expect(disabledMenuItem.text()).toBe('foo');
});

it('does not render a hidden menu item', () => {
  const wrapper = mount(
    <Menu onChange={onChangeMockEvent}>
      <MenuItem label="foo" hidden />
      <MenuItem label="bar" />
    </Menu>,
  );
  const menuItems = wrapper.find('.el-menu__item');
  expect(menuItems).toHaveLength(1);
});

it('does not render a hidden menu section', () => {
  const wrapper = mount(
    <Menu onChange={onChangeMockEvent}>
      <MenuSection hidden>
        <MenuItem label="foo" />
        <MenuItem label="bar" />
      </MenuSection>
    </Menu>,
  );
  const menuSection = wrapper.find('.el-menu__section');
  expect(menuSection).toHaveLength(0);
});

it('trigger onChange when clicking a menu item', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <Menu path="/" pathname="/" onChange={mockEvent}>
      <MenuItem path="/foo" label="foo" />
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  const menuItems = wrapper.find('.el-menu__item');
  menuItems.at(0).simulate('click');
  expect(mockEvent).toHaveBeenCalled();
});

it('trigger onChange with correct value when clicking a menu item', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <Menu path="/" pathname="/" onChange={mockEvent}>
      <MenuItem path="/foo" label="foo" />
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  const menuItems = wrapper.find('.el-menu__item');
  menuItems.at(1).simulate('click');
  expect(mockEvent).toHaveBeenCalledWith('/bar');
});

// Snapshot
it('renders the menu correctly when multiple props are set', () => {
  const wrapper = mount(
    <Menu path="/" pathname="/foo" onChange={onChangeMockEvent}>
      <MenuItem path="/foo" label="foo" asRoot>
        <MenuItem path="/foo/nested" label="nested" />
      </MenuItem>
      <MenuItem path="/bar" label="bar" />
    </Menu>,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
