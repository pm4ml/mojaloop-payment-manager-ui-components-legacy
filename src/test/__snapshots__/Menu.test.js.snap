// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the menu correctly when multiple props are set 1`] = `
<Menu
  onChange={[MockFunction]}
  path="/"
  pathname="/foo"
>
  <div
    className="mb-element el-menu"
  >
    <MenuItemsGroup
      key="0"
    >
      <div
        className="el-menu__section-items"
      >
        <MenuItem
          asRoot={false}
          back={false}
          disabled={false}
          hidden={false}
          key=".0"
          label="nested"
          namedIcon={false}
          onClick={[MockFunction]}
          partial={false}
          path="/foo/nested"
          pathname="/foo"
          size={14}
        >
          <div
            className="el-menu__item"
            onClick={[Function]}
            role="presentation"
          >
            nested
          </div>
        </MenuItem>
      </div>
    </MenuItemsGroup>
  </div>
</Menu>
`;
